### ISSUE TITLE
introduce `on-deprecate`

### ISSUE THREAD

**Issue by aconrad** [SurveyMonkey/defrost#5]
I would like to let the user have more fine-grained control over the behavior of defrost. Here I'm introducing a new keyword `on-deprecate` which would take either `error`, `warn` or `ignore` as a value.

```yaml
---
requirements:
- requirement: foo==2.0
  reason: please upgrade!
  on-deprecate: error|warn
```

The default value would be `error` if `on-deprecate` is not present.

* `error`: exit code 1 if package doesn't meet requirement
* `warn`: exit code 0 if package doesn't meet requirement (unless another package doesn't meet its requirement and is set to `on-deprecate: error`)

It would also be appropriate to introduce the command line option `-z` or `--zero` to force the exit code to be 0 regardless of the `on-deprecate` value.

@sudarkoff @msabramo

**Comment by aconrad** [comment]
I think I prefer the term `severity` rather than `on-deprecate`. It's a one-worder and is more targeted to the importance of the unmet requirement whereas `on-deprecate` invites to take a further action.

**Comment by sudarkoff** [comment]
Yeah, I like `severity`!

**Comment by sudarkoff** [comment]
And I like `--zero`! And I'd like a similar option for when you want to be very strict and treat warnings as errors. Maybe something like `--error` and `--no-error`? Or something more descriptive.

**Comment by aconrad** [comment]
Fixed by #8.
